### ISSUE TITLE
"No entities error"

### ISSUE THREAD

**Issue by Rockwell70** [beekeeper-studio/beekeeper-studio#527]
I've connected Beekeeper to multiple Sqlite databases, but one in particular is causing me problems. It connects successfully, but shows no entities. The database actually has 10 tables, and they appear in other Sqlite db browsers.

**Version Information [might be relevant to your issue]**
Operating System: Windows 10
App Version [help -> about]: 1.9.1

**Comment by rathboma** [comment]
Huh, that's weird. Are there any errors in the console? (help -> show developer tools -> console) or in the log file? (https://docs.beekeeperstudio.io/troubleshooting/#enable-debug-mode-and-collect-logs).

If you are able to share the database with me, that would of course be the easiest way to debug it :-)

**Comment by Rockwell70** [comment]
I’d share it if I could but it’s got proprietary info in it. I could strip out all the data and just keep the tables if that would help.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by craigosborne** [comment]
I've also had this issue.
In the console I get the error 'SQLITE_ERROR neear UPDATE'. The database I have an issue with has a table called update. I suspect the name of the table is causing the issue.

OS: Windows 10
version: 1.9.1

**Comment by Rockwell70** [comment]
Interesting. I get no errors. It’s as if BK thinks it’s loaded an empty database.

JS

On Jan 27, 2021, at 00:45, craigosborne <notifications@github.com> wrote:

﻿

I've also had this issue.
In the console I get the error 'SQLITE_ERROR neear UPDATE'. The database I have an issue with has a table called update. I suspect the name of the table is causing the issue.

OS: Windows 10
version: 1.9.1

**Comment by rathboma** [comment]
Yes that's why!!! Poor escaping. Easy fix, thank you so much for debugging

**Comment by IceWreck** [comment]
I have the same issue. I dont have any table name containing "update" though.

This is what i get in the javascript console
```
chunk-vendors.475a633a.js:2332 Error: SQLITE_ERROR: unrecognized token: "2018_lib"
--> in Database#all('PRAGMA table_info(2018_lib);', undefined, [Function])
    at app.1703b08a.js:1
    at new Promise (<anonymous>)
    at n (app.1703b08a.js:1)
    at a (app.1703b08a.js:1)
    at Database.<anonymous> (app.1703b08a.js:1)
```

**Comment by Amar1729** [comment]
To help anybody else finding this - I'm seeing this error as well for table names containing characters such as hyphens or underscores.

```
Uncaught {in promise) Error:
SQLITE ERROR: near "-": syntax error
=> in Database#all('PRAGMA table_info(table-1);', undefined,
[Function])
    at app.3a32bb91.js:1
    at new Promise (<anonymous>)
    at n (app.3a32bb91.js:1)
    at a (app.3a32bb91.js:1)
    at async Database.<anonymous> (app.3a32bb91.js:1)
```

sample sql to generate a bad db:
```
CREATE TABLE IF NOT EXISTS "table-1" (
    id integer PRIMARY_KEY,
    time real NOT NULL,
    some_text text
);

CREATE TABLE IF NOT EXISTS "table_1" (
    id integer PRIMARY_KEY,
    time real NOT NULL,
    some_text text
);
```

**Comment by gersh0m** [comment]
Same Issue with table name:
```references```
As a workaround I had to change table name.

**Comment by Khanali00-lab** [comment]
I have the same issue. When i import file it says there are no entities in C:\Users\name... I have tried change the file type (CSV, pdf, db, etc) but same result. I have struggled with this 2 days now. I have checked youtube and other websites but they were not helpful.
